### PR TITLE
CPLP-836 My Business Applications

### PR DESCRIPTION
**Pull request by wira-mms**

### CODE REVIEW

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs [hunk 1]
```diff
@@ -41,6 +41,29 @@ public IAsyncEnumerable<AppViewModel> GetAllActiveAppsAsync([FromQuery] string?
...
+        /// <response code="400">If sub claim is empty/invalid.</response>
+        [HttpGet]
+        [Route("business")]
+        [Authorize(Roles = "view_apps")]
+        [ProducesResponseType(typeof(IAsyncEnumerable<BusinessAppViewModel>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        public ActionResult<IAsyncEnumerable<BusinessAppViewModel>> GetAllBusinessAppsForCurrentUserAsync()
+        {
+            var userId = GetIamUserIdFromClaims();
+            if (string.IsNullOrWhiteSpace(userId))
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs:59]: No explicit mapping to Statuscodes - businesslogic should throw ArgumentException (which will be mapped to BadRequest with propper json error-response body by the GeneralHttpErrorHandler in the middleware).

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs (hunk 1 above)
Comment (jjeroch) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs]: @ntruchsess pagination does not need to get considered in that scenario due to a very low amount of elements

**Comment by jjeroch** [comment]
FYI I have run through the code as well, from what I could see I dont have any finding so far. I am planning to run a test later still before starting FE integration

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs (hunk 1 above)
Comment (wira-mms) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs]: https://github.com/catenax-ng/product-portal-backend/pull/54/commits/157492e5355f379831ce90e80dcc33b4ecc2faca

**Comment by ntruchsess** [comment]
@jjeroch  : shall this endpoint implement pagination?

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs [hunk 2]
```diff
@@ -41,6 +41,29 @@ public IAsyncEnumerable<AppViewModel> GetAllActiveAppsAsync([FromQuery] string?
...
+    /// <response code="400">If sub claim is empty/invalid.</response>
+    [HttpGet]
+    [Route("business")]
+    [Authorize(Roles = "view_apps")]
+    [ProducesResponseType(typeof(IAsyncEnumerable<BusinessAppViewModel>), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+    public ActionResult<IAsyncEnumerable<BusinessAppViewModel>> GetAllBusinessAppsForCurrentUserAsync()
+    {
+        var userId = GetIamUserIdFromClaims();
+        if (string.IsNullOrWhiteSpace(userId))
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs:59]: please move this check and throw of ArgumentException into method GetIamUserIdFromClaims(), then you can remove all this duplicated code

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs (hunk 2 above)
Comment (wira-mms) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppsController.cs]: https://github.com/catenax-ng/product-portal-backend/pull/54/commits/d71c38334a0b481631d54a1572264f851de73e4a

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**